lib/text_client.rb

```diff
@@ -0,0 +1,110 @@
+require 'game'
+
+class TextClient
+  def initialize()
+    @current_player = :player_one
+  end
+
+  def start
+    game = Game.new()
+    play(game) if game.state == :ready
+    :finished
+  end
+
+  private
+
+  def play(game)
+    while game.state != :game_over
+      update_ui(game)
+      move = get_next_move(@current_player)
+      game = game.make_move(@current_player, move)
+      swap_current_player if game.state == :ok
+    end
+    update_ui(game)
+  end
+
+  def update_ui(game)
+    print_game_state(game.state, game.result) if game.state == :ready
+    print_board(game.board_state)
+    print_game_state(game.state, game.result) unless game.state == :ready
+    print_available_positions(game.available_positions) unless game.state == :game_over
+    print_get_next_move_prompt(@current_player) unless game.state == :game_over
+  end
+
+  def print_board(board)
+    puts ""
+    draw_row(board[0], board[1], board[2])
+    draw_spacer_row
+    draw_row(board[3], board[4], board[5])
+    draw_spacer_row
+    draw_row(board[6], board[7], board[8])
+  end
+
+  def draw_row(a, b, c)
+    puts " #{marker_for(a)} | #{marker_for(b)} | #{marker_for(c)} "
+  end
+
+  def marker_for(id)
+    case id
+    when 1
+      "X"
+    when 2
+      "O"
+    else
+      " "
+    end
+  end
+
+  def draw_spacer_row
+    puts '-----------'
+  end
+
+  def print_game_state(state, result)
+    text = case state
```

Comment:
Could this entire state series be extracted out? Seems like it'd be easier to deal with

Reply:
That's a good suggestion. Also means I can test for key bits of text easily too. 👍